main/params.go

```diff
@@ -677,21 +680,43 @@ func setNodeConfig(v *viper.Viper) error {
 	Config.EnableCrypto = v.GetBool(signatureVerificationEnabledKey)
 
 	// Coreth Plugin
-	corethConfigString := v.GetString(corethConfigKey)
-	if corethConfigString != defaultString {
-		corethConfigValue := v.Get(corethConfigKey)
-		switch value := corethConfigValue.(type) {
-		case string:
-			corethConfigString = value
-		default:
-			corethConfigBytes, err := json.Marshal(value)
-			if err != nil {
-				return fmt.Errorf("couldn't parse coreth config: %w", err)
-			}
-			corethConfigString = string(corethConfigBytes)
+	// TODO: deprecate
+	corethConfigValue := v.Get(corethConfigKey)
+	corethConfigBytes, err := utils.MarshalBytes(corethConfigValue)
+	if err != nil {
+		return fmt.Errorf("could not parse coreth config: %w", err)
+	}
+	Config.CorethConfig = string(corethConfigBytes)
+
+	// ChainConfigs
+	rawChainConfigs, err := utils.MarshalMapSlice(v.Get(chainConfigsKey))
+	if err != nil {
+		return fmt.Errorf("chain configs are not an array of JSON objects: %w", err)
+	}
+
+	Config.ChainConfigs = map[ids.ID]chains.ChainConfig{}
+	for i, rawChainConfig := range rawChainConfigs {
+		rawChainID, ok := rawChainConfig[chainIDKey]
+		if !ok {
+			return fmt.Errorf("could not parse chainID from chain config %d", i)
+		}
+
+		chainID, ok := rawChainID.(string)
+		if !ok {
+			return fmt.Errorf("chainID `%+v` is not a string in chain config %d", rawChainID, i)
+		}
+
+		id, err := ids.FromString(chainID)
+		if err != nil {
+			return fmt.Errorf("could not parse chainID %s: %w", chainID, err)
+		}
+
+		config := chains.ChainConfig{}
+		if err := utils.SetByteSlices(rawChainConfig, &config); err != nil {
+			return fmt.Errorf("could not parse chain config for %s: %w", chainID, err)
 		}
+		Config.ChainConfigs[id] = config
```

Comment:
Should we error if the same chain is specified multiple times?

Reply:
Chain here being either a chainID or an alias that maps to that chainID, right?

Reply:
unless, we wanted to have the specification of a chainID override any aliases